nixos/modules/services/misc/sonarr.nix

```diff
@@ -7,6 +7,20 @@
 }:
 let
   cfg = config.services.sonarr;
+
+  settingsFormat = pkgs.formats.xml { };
+  settingsDefault = {
+    Port = 8989;
+    BindAddress = "*";
+    AuthenticationMethod = "None";
+    UpdateMechanism = "external";
+  };
+  settingsCombined = settingsDefault // cfg.settings;
```

Comment:
settingsCombined can be removed now and replaced with `cfg.settings` everywhere, I think.

Reply:
i think we can not. If we would do so, and define it as "default". And then the user provides the settings. It will not get "merged" (`//`) together. instead the default gets replaced with the settings provided.
Some settings like `UpdateMechanism` make sense to keep them if not provided in another way.

Reply:
Rather than have `settingsCombined`, you can do something like this:

```nix
      config.services.sonarr.settings = lib.mapAttrs (_: lib.mkDefault) settingsDefault;
```

Reply:
What about using a [freeform submodule](https://nixos.org/manual/nixos/stable/index.html#sec-freeform-modules) here?

Reply:
is implemented.